packages/wrangler/templates/startDevWorker/ProxyWorker.ts

```diff
@@ -127,6 +130,8 @@ export class ProxyWorker implements DurableObject {
 			// if we decide to await, we should include a timeout (~100ms) in case the user worker has long-running/parellel requests
 			void fetch(userWorkerUrl, new Request(request, { headers }))
 				.then((res) => {
+					rewriteUrlRelatedHeaders(headers, innerUrl, outerUrl);
```

Comment:
Ah yes both directions! This is clever. Now the browser will have no idea that the URL was faked inside the worker either 😄 

Small bug/typo:
```suggestion
					rewriteUrlRelatedHeaders(res.headers, innerUrl, outerUrl);
```

But you might not be able to mutate response headers without:
```suggestion
					res = new Response(res);
					rewriteUrlRelatedHeaders(res.headers, innerUrl, outerUrl);
```

Reply:
That's not a typo! It's a bona fide mistake! Thanks for spotting that.

Reply:
I also think this is a "new"-ish feature that should get its own changeset and documentation. It's really cool that a dev could write ``Response.redirect(`${request.origin}/redirect/path`)`` and that would "just work"™️ in `wrangler dev [--remote]` – we should shout from the rooftops!

Reply:
Strictly this was already part of the old proxy: https://github.com/cloudflare/workers-sdk/blob/d637bd59a8ea6612d59ed4b73e115287615e617d/packages/wrangler/src/dev/proxy.ts#L70